src/components/createConnect.js

```diff
@@ -77,6 +77,8 @@ export default function createConnect(React) {
     }
 
     return function wrapWithConnect(WrappedComponent) {
+      let isStateless = !(WrappedComponent.prototype && WrappedComponent.prototype.isReactComponent) || !('prototype' in WrappedComponent);
```

Comment:
I'm not sure if this is a major concern, but this method of detecting stateful components isn't compatible with React < 0.14. Stateful components created in React 0.13 will be flagged as stateless as well. We had a similar situation at https://github.com/gajus/react-css-modules/issues/40